docs/theming.md

```diff
@@ -2,37 +2,92 @@
 
-```bash
-npm install -S emotion-theming
+We recommend creating a file which exports a ThemeProvider component and a useTheme hook.
+
+```jsx
+import { createContext, useContext } from 'react'
+
+let defaultTheme = {
+  colors: {
+    primary: 'hotpink',
+    secondary: 'green'
+  }
+}
+
+let ThemeContext = createContext(defaultTheme)
+
+export let ThemeProvider = ThemeContext.Provider
+
+export let useTheme = () => useContext(ThemeContext)
 ```
 
-Add `ThemeProvider` to the top level of your app and access the theme with `props.theme` in a styled component or provide a function that accepts the theme as the css prop. The api is laid out in detail [in the documentation](https://emotion.sh/docs/emotion-theming).
+The theme can be used with css prop like this.
 
 ```jsx
-// @live
 /** @jsx jsx */
 import { jsx } from '@emotion/core'
+import { useTheme } from './theme'
+
+let Button = props => {
+  let theme = useTheme()
+  return (
+    <button
+      css={{
+        color: theme.colors.primary
+      }}
+      {...props}
+    />
+  )
+}
+
+let SecondaryButton = props => {
+  let theme = useTheme()
+  return (
+    <button
+      css={{
+        color: theme.colors.secondary
+      }}
+      {...props}
+    />
+  )
+}
+```
+
+Or with the styled api like this
+
+```jsx
 import styled from '@emotion/styled'
-import { ThemeProvider } from 'emotion-theming'
+import { useTheme } from './theme'
 
-const theme = {
-  colors: {
-    primary: 'hotpink'
+let Button = styled.button(() => {
+  let theme = useTheme()
+
+  return {
+    color: theme.colors.primary
   }
-}
+})
 
-const SomeText = styled.div`
-  color: ${props => props.theme.colors.primary};
+let SecondaryButton = styled.button`
+  color: ${() => useTheme().colors.secondary};
 `
-
-render(
-  <ThemeProvider theme={theme}>
-    <SomeText>some text</SomeText>
-    <div css={theme => ({ color: theme.colors.primary })}>
-      some other text
-    </div>
-  </ThemeProvider>
-)
 ```
+
+## Why don’t we have a built in theming API?
+
+Theming in emotion v9 and earlier was inspired by theming in styled-components which was created before React’s new Context and Hooks existed. This meant having access to a value through context was novel but with React's new features, problems with that API can be solved and having a theming API built in is unnecessary
+
+### Type Safety
+
+The previous theming API was extremely hard to statically type. By using context and hooks directly, a default value can be provided so type systems can guarantee that there's always a theme that can be accessed.
+
+### Global Namespace
+
+Since there’s was a single object which contained the theme, if a component from npm or somewhere wanted to use theming, it had to choose a unique name store it’s theme in which meant name conflicts could occur. By directly using the Context API, this problem doesn't exist.
```

Comment:
This shouldnt be relevant point here because it refers to legacy context and it's not a problem with new context APi.

Reply:
oh, i have kinda skipped the part where you mention libraries built on top of emotion 😅